### ISSUE TITLE
Change default column name when reading from file OR change `geometry` property to `geometry_column`

### ISSUE THREAD

**Issue by nickeubank** [geopandas/geopandas#294]
As it stands, when a new file is read it, the column with shapely objects is named "geometry". 

The problem is that we also talk about the "geometry column" -- the GeoSeries with special standing in a GeoDataFrame. 

But if the user ever makes a new GeoSeries and sets it as the "geometry column" with `set_geometry`, then suddenly there's a column named "geometry" and also a different column associated with the "gdf.geometry" attribute. This seems a recipe for confusion. 

I propose either (a) renaming the `geometry` property something like `geometry_column`, or (b) changing the default name for a column to something new ("shapes"? "geo"? "Fred"? Open to suggestions.)

**Comment by jwass** [comment]
Hi @nickeubank,
Isn't this still the same as [@jorisvandenbossche's comment](https://github.com/geopandas/geopandas/issues/291#issuecomment-202150794)?

If you need the specific column named 'geometry' you'll need `df['geometry']` otherwise `df.geometry` gives the "geometry column". It's just like the distinction between `df.mean` and `df['mean']`.

**Comment by nickeubank** [comment]
Sorry, I wasn't very clear in my post: from a technical perspective, it
works fine. My concern is that i think it's bad design and that it's
potentially very confusing for new users -- and unnecessarily so. I think a
name change could make the whole thing much more intuitive.

**Comment by jorisvandenbossche** [comment]
@nickeubank a name change to for example `geo` or `shapes` is exactly the same design, and will have the same problem for people with other columns called 'geo' or 'shape', so I don't really see the benefit of that.

Of course something like 'geometry_column' will be less likely to be used as a casual column name, but is also a lot less inconvenient in using in operations.

**Comment by nickeubank** [comment]
@jorisvandenbossche It doesn't make the problem go away entirely, true. But as it stands, the default state of a geodataframe (the state it takes on when you import from a file) is the most confusing state possible for new users -- the same term "geometry" is used to refer to two different things (the "active" column and the name of the column of shapely objects). To make matters worse, in the default state they are actually both pointing to the same series! 

Put yourself in the shoes of a new user. You read a file in from disk. You know there's a concept of an "active column" called geometry. If you type `gdf.geometry`, you get the column *named* geometry back. So you assume "oh, ok, the column named geometry is the "geometry column"". 

This confusion can easily go further. The property that identifies the "active column" (see, just to write this post I need new terms!) is just the column name, so if the user, assuming the column named geometry is the "active column" tries to act on that by swapping column names to change the "active column", turns out that works fine, compounding the confusion. If you do:

    gdf['new_series'] = gdf.geometry.centoid
    gdf = gdf.rename(columns={'geometry': 'new_series', 'new_series': 'geometry'})
    gdf.plot()

the result will be that the centroids are plotted. (I know, because this way my initial confusion!)  

If the default state of a geodataframe loaded from a file is to call the column of shapely objects something else ("Joe"), then it's clear right off the bat it's clear that the active column and column names are distinct concepts. Yes, the user could wander into a situation where they name a column "geometry", but by then they'd be familiar with these being different concepts. 

Moreover, it's easier to explain if there are different names for things. Just to write this I've had to adopt the term "active column" (which, incidentally, I think is actually a much better than than "geometry").  

So in summary: yes, the problem that attributes and columns can have the same name is a constant in ``pandas``. But by having a default state that (a) uses the same term to refer to two things that are conceptually distinct, but (b) which in the default state are actually not distinct in their specifics, I think new users will be unnecessarily and very understandably confused.

**Comment by micahcochran** [comment]
On minor thing in favor of renaming this to `geometry_column` (15 bytes), is that shapefile DBF's have the 11 byte limit, so data from shapefile sources will never have this name.

Most importantly renaming `geometry` breaks API backward compatibility.

**Comment by nickeubank** [comment]
@micahcochran yup, good call on both fronts. 

The reverse compatibility is biggest problem to be sure. I think that if we do want to do this, that would suggest it's better to rename default column than change the `geometry attribute` -- that way old code would (a) always break not fail silently (making problem obvious), and (b) could be fixed if you just append `.rename(columns={'NEW_DEFAULT': 'geometry'})` to the end of a `read_file` command. 

Relatedly: after some thought, I think `file_shapes` makes a reasonable default name (though it's 11 bytes, so doesn't achieve @micahcochran's suggestion of exceeding 11 bytes to avoid collisions). `geometry_column` is different from `geometry`, but isn't actually that more more informative a name.
